### PR TITLE
Registered tokens json changes

### PR DESCRIPTION
**Pull request by piotrkosecki**
Should be merged after #1019 

Updated registered tokens table - now it's inserted from the JSON file(@anonymoussprocket please provide some viable data for it), it also can update table from URL(as it's based on the branch for ticket #1018 )

### CODE REVIEW

File: conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosDatabaseOperations.scala [hunk 1]
```diff
@@ -848,7 +860,8 @@ object TezosDatabaseOperations extends ConseilLogSupport {
       db: Database,
       table: TableQuery[A],
       network: String,
-      separator: Char = ','
+      separator: Char = ',',
+      clean: Boolean = false
```
Comment (vishakh) [inline conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosDatabaseOperations.scala:864]: Make more descriptive or document

File: conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosDatabaseOperations.scala [hunk 2]
```diff
@@ -868,6 +881,62 @@ object TezosDatabaseOperations extends ConseilLogSupport {
         Future.successful(List.empty -> None)
     }
 
+  /** Reads json file for registered tokens */
+  def readJsonFile(network: String): List[RegisteredToken] = {
```
Comment (vishakh) [inline conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosDatabaseOperations.scala:885]: Please rename

File: conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosDatabaseOperations.scala [hunk 3]
```diff
@@ -868,6 +881,62 @@ object TezosDatabaseOperations extends ConseilLogSupport {
...
+    import io.circe.parser.decode
+    import RegisteredTokensFetcher.decoder
+
+    val file = getClass.getResource(s"/registered_tokens/$network.json")
+    val content = Source.fromFile(file.toURI).getLines.mkString
+    decode[List[RegisteredToken]](content) match {
+      case Left(error) =>
+        logger.error(s"Something wrong with registered tokens file $error")
```
Comment (vishakh) [inline conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosDatabaseOperations.scala:893]: Lorre should shut down if the content is malformed or we will never realize we don't have proper data.

File: conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosIndexer.scala [hunk 4]
```diff
@@ -342,39 +349,23 @@ object TezosIndexer extends ConseilLogSupport {
...
-
-          }
-
-        Await.result(tokenContractsFuture, 5.seconds)
+      (
+        TezosDb.initTableFromCsv(db, Tables.KnownAddresses, selectedNetwork),
+        TezosDb.initTableFromCsv(db, Tables.BakerRegistry, selectedNetwork),
+        TezosDb.initRegisteredTokensTableFromJson(db, selectedNetwork)
+      ).mapN {
+        case (_, _, _) => ()
```
Comment (vishakh) [inline conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosIndexer.scala:360]: Graceful shutdown if *any* processing doesn't happen.

File: conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/RegisteredTokensFetcher.scala [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      isTzip16: Boolean,
+      isNft: Boolean,
+      metadataType: Option[String],
+      metadataBigMapId: Option[Int],
+      metadataBigMapType: Option[String],
+      metadataPath: Option[String]
+  )
+
+  /** Updates Registered tokens table from URL */
+  def updateRegisteredTokens(
```
Comment (vishakh) [inline conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/RegisteredTokensFetcher.scala:44]: Shut down if it goes wrong.

**Review by vishakh — CHANGES_REQUESTED** [review]

File: conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/RegisteredTokensFetcher.scala (hunk 5 above)
Comment (piotrkosecki) [reply conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/RegisteredTokensFetcher.scala]: ok, done

File: conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosIndexer.scala (hunk 4 above)
Comment (piotrkosecki) [reply conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosIndexer.scala]: ok, done

File: conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosDatabaseOperations.scala (hunk 2 above)
Comment (piotrkosecki) [reply conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosDatabaseOperations.scala]: renamed

File: conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosDatabaseOperations.scala (hunk 3 above)
Comment (piotrkosecki) [reply conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosDatabaseOperations.scala:893]: shutting down

File: conseil-lorre/src/main/resources/application.conf [hunk 6]
```diff
@@ -84,6 +86,13 @@ lorre {
     update-size: 16 # amount of rights we update after Lorre syncs
   }
 
+  token-contracts {
+    url: "https://61012ab14e50960017c29c53.mockapi.io/registered_tokens/registered_tokens"
+    url: ${?CONSEIL_LORRE_REGISTERED_TOKENS_URL}
+    initial-delay: 2 minutes
+    interval: 1 minute
```
[the remark this answers is not in the document]
Comment (piotrkosecki) [reply conseil-lorre/src/main/resources/application.conf]: changed

**Review by vishakh — APPROVED** [review]

**Pull request merged**